helix-view/src/editor.rs

```diff
@@ -1460,10 +1460,7 @@ impl Editor {
             self.ensure_cursor_in_view(view_id);
 
             // Update jumplist selections with new document changes.
-            for (view, _focused) in self.tree.views_mut() {
-                let doc = doc_mut!(self, &view.doc);
-                view.sync_changes(doc);
-            }
+            self.sync_views();
```

Comment:
We don't need to sync unfocused docs here so using a shared function does more work than we need. Let's eliminate the shared `sync_views` function and instead sync the views as we create the items for the file picker (currently [this block](https://github.com/helix-editor/helix/blob/e72ad1e731877570c6f728e1490485e4b31524a5/helix-term/src/commands.rs#L2570-L2578))

Reply:
because the `new meta` closure captures `&cx.editor`, it is impossible to also use `&mut cx.editor` to create the `&mut Doc' that is needed to sync the view with a doc so i have opted to sync the views earlier in the function.